sdk/communication/azure-communication-rooms/azure/communication/rooms/_models/_models.py

```diff
@@ -3,52 +3,128 @@
          following document for additional information on Raw ID. :code:`<br>`
          https://learn.microsoft.com/azure/communication-services/concepts/identifiers?pivots=programming-language-rest#raw-id-representation.
+        :paramtype communication_identifier: str or ~azure.communication.rooms.CommunicationIdentifier
         :keyword role: The role of a room participant. The default value is Attendee. Known
          values are: "Presenter", "Attendee", and "Consumer".
         :paramtype role: str or ~azure.communication.rooms.models.ParticipantRole
         """
-        super().__init__(**kwargs)
-        self.communication_identifier = communication_identifier
-        self.role = role or ParticipantRole.ATTENDEE
-
+    @overload
+    def __init__(self, mapping: Mapping[str, Any]) -> None:
+        """
+        :param mapping: raw JSON to initialize the model. Positional only.
+        :type mapping: Mapping[str, Any]
+        """
+    
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        if len(args) == 1:
+            self.communication_identifier = identifier_from_raw_id(args[0]['rawId'])
+            self.role = args[0]['role']
+        elif args:
+            raise TypeError(f"RoomParticipant.__init__() takes 2 positional arguments but {len(args) + 1} were given.")
+        else:
+            identifier = kwargs['communication_identifier']
+            if isinstance(identifier, str):
+                self.communication_identifier = identifier_from_raw_id(identifier)
+            else:
+                self.communication_identifier = identifier
+            self.role = kwargs.get('role', ParticipantRole.ATTENDEE)
 
     def __eq__(self, other):
         if isinstance(other, self.__class__):
             return self.__dict__ == other.__dict__
         return False
 
-    @classmethod
-    def _from_generated(cls, room_participant):
-        return cls(
-            communication_identifier=identifier_from_raw_id(room_participant.raw_id),
-            role=room_participant.role
-        )
+
+class CommunicationRoom:
+    """The meeting room.
+
+    All required parameters must be populated in order to send to Azure.
+
+    :ivar id: Unique identifier of a room. This id is server generated. Required.
+    :vartype id: str
+    :ivar created_at: The timestamp when the room was created at the server. The timestamp is in
+     RFC3339 format: ``yyyy-MM-ddTHH:mm:ssZ``. Read-only.
+    :vartype created_at: ~datetime.datetime
+    :ivar valid_from: The timestamp from when the room is open for joining. The timestamp is in
+     RFC3339 format: ``yyyy-MM-ddTHH:mm:ssZ``.
+    :vartype valid_from: ~datetime.datetime
+    :ivar valid_until: The timestamp from when the room can no longer be joined. The timestamp is
+     in RFC3339 format: ``yyyy-MM-ddTHH:mm:ssZ``.
+    :vartype valid_until: ~datetime.datetime
+    """
+
+    @overload
+    def __init__(
```

Comment:
DO we still need this overload since we removed the overload for `update_room`?

Reply:
Yes we need these for the forwards compatibility of the models :)